xformers/components/attention/csrc/cpu/attention.cpp

```diff
@@ -90,15 +92,18 @@ void attention_kernel(
         for (int64_t k = 0; k < K; k++) {
           oo[k] = buf[k] / s_prime;
         }
+        if (compute_logsumexp)
+          logsumexp[i][j] = m_prime + std::log(s_prime);
```

Comment:
ah, I didn't think of that, this is why your backward is so fast, and it does not weight that much actually (one per line, not the whole attention map)

Reply:
yeah, keeping this temporary around made the backward kernel faster and also easier to implement, so I decided that it was worth the extra memory.

Also, in a follow-up PR I'll avoid allocating the `logsumexp` buffer in the function if `compute_logsumexp` is `false`, so that this is only a memory price to pay during training